kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/marshalling/json/JSONMarshaller.java

```diff
@@ -503,20 +502,34 @@ public ExtendedJaxbAnnotationIntrospector(List<NamedType> customClasses, ObjectM
 
         @Override
         public List<NamedType> findSubtypes(Annotated a) {
-            List<NamedType> base = super.findSubtypes(a);
-
             List<NamedType> complete = new ArrayList<NamedType>();
-            if (base != null) {
-                complete.addAll(base);
-            }
             if (customClasses != null) {
-                for (NamedType namedType  : customClasses) {
+                for (NamedType namedType : customClasses) {
                     Class<?> clazz = namedType.getType();
                     if (!a.getRawType().equals(clazz) && a.getRawType().isAssignableFrom(clazz)) {
                         complete.add(namedType);
                     }
                 }
             }
+
+            XmlElements elements = findAnnotation(XmlElements.class, a, false, false, false);
+            if (elements != null) {
+                for (XmlElement elem : elements.value()) {
```

Comment:
can't we check jsonHandleXmlAnyElementsNames before the loop, so we can avoid unnecessary processing (I know that over time JIT will do the job... but still)

Reply:
it won't give the same outcome.